python/dolfinx/mesh.py

```diff
@@ -80,89 +102,109 @@ def locate_entities_boundary(mesh: cpp.mesh.Mesh,
 
     """
 
-    return cpp.mesh.locate_entities_boundary(mesh, dim, marker)
+    return _cpp.mesh.locate_entities_boundary(mesh, dim, marker)
 
 
 _uflcell_to_dolfinxcell = {
-    "interval": cpp.mesh.CellType.interval,
-    "triangle": cpp.mesh.CellType.triangle,
-    "quadrilateral": cpp.mesh.CellType.quadrilateral,
-    "tetrahedron": cpp.mesh.CellType.tetrahedron,
-    "hexahedron": cpp.mesh.CellType.hexahedron
+    "interval": CellType.interval,
+    "triangle": CellType.triangle,
+    "quadrilateral": CellType.quadrilateral,
+    "tetrahedron": CellType.tetrahedron,
+    "hexahedron": CellType.hexahedron
 }
 
 _meshtags_types = {
-    numpy.int8: cpp.mesh.MeshTags_int8,
-    numpy.int32: cpp.mesh.MeshTags_int32,
-    numpy.int64: cpp.mesh.MeshTags_int64,
-    numpy.double: cpp.mesh.MeshTags_double
+    np.int8: _cpp.mesh.MeshTags_int8,
+    np.int32: _cpp.mesh.MeshTags_int32,
+    np.int64: _cpp.mesh.MeshTags_int64,
+    np.double: _cpp.mesh.MeshTags_double
 }
 
 
-def refine(mesh, cell_markers=None, redistribute=True):
-    """Refine a mesh"""
+def refine(mesh: Mesh, cell_markers: _cpp.mesh.MeshTags_int8 = None, redistribute: bool = True):
+    """Refine a mesh
+
+    Parameters
+    ----------
+    mesh
+        The mesh from which to build a refined mesh
+    cell_markers
+        Optional argument to specify which cells should be refined. If not supplied uniform refinement.
```

Comment:
This should be "entity markers" not "cell markers". The fundamental operation of refinement is on edges, so if you supply cells, it will refine all the edges of the cell - but you can also just supply the edges.

Reply:
I'll fix that when merging in: https://github.com/FEniCS/dolfinx/issues/1834 afterwards